application/plugin_base/item_list.py

```diff
@@ -1647,6 +1665,19 @@ def destroy_providers(self):
 		for path, provider in self._providers.items():
 			provider.release_archive_handle()
 
+	def create_intermediate_archive_provider(self, path=None):
```

Comment:
I understand the purpose, but am not really liking the way it works. It's a single purpose method for only specific use cases. There has to be a better way. Issue can be partially mitigated by getting provider from the opposite object, from which we are inheriting path anyway.

That leaves issue of restoring path when restarting Sunflower, but I feel that should be a separate function.